### PR TITLE
Add VCR.py json-serialized tests

### PR DESCRIPTION
**Pull request by vshlapakov**
The PR adds additional tests formed with json serialization based on:
- new vcrpy accept-header matcher to distinguish messagepack-marked requests from others
- additional fixture `json_and_msgpack` to parametrize some of the existing tests
- json-serialized tests being put to separate vcrpy cassettes, named with `-json.gz` suffix
- pytest `--disable-msgpack` option to disable msgpack tests (same if msgpack is unavailable)

I have some doubts if we need to add Travis configuration to run all the tests with uninstalled msgpack to make sure everything works as expected w/o msgpack at all, let's discuss it if needed.

Please, review.

### CODE REVIEW

**Comment by chekunkov** [comment]
> I have some doubts if we need to add Travis configuration to run all the tests with uninstalled msgpack to make sure everything works as expected w/o msgpack at all

why not?

**Comment by vshlapakov** [comment]
> why not?

@chekunkov  I had to add some new tests and corresponding cassettes for oldy hubstorage client, now satisfied with results, please review when you have a minute.

File: .travis.yml [hunk 1]
```diff
@@ -2,17 +2,29 @@ language: python
 matrix:
   include:
     - python: 2.7
-      env: TOXENV=py27
+      env: TOXENV=py27-json
```
Comment (chekunkov) [inline .travis.yml]: note that this breaks `$TOXENV == py27` check in deploy section

**Comment by vshlapakov** [comment]
@chekunkov Let me know if there's anything else worrying you so we can proceed with the PR.

**Pull request merged**